services/core-web/src/actionCreators/documentActionCreator.js

```diff
@@ -50,20 +53,25 @@ export const generateNoticeOfWorkApplicationDocument = (
       createRequestHeader()
     )
     .then((response) => {
-      const token = { token: response.data.token };
-      const docWindow = window.open(
-        `${ENVIRONMENT.apiUrl + API.RETRIEVE_CORE_DOCUMENT(token)}`,
-        "_blank"
-      );
-      docWindow.onbeforeunload = () => {
-        notification.success({
-          message,
-          duration: 10,
+      const params = { token: response.data.token, return_record: "true" };
+      return CustomAxios()
+        .get(`${ENVIRONMENT.apiUrl + API.DOCUMENT_GENERATION(params)}`, createRequestHeader())
+        .then((response) => {
+          const mineDocument = response.data.mine_document;
+          notification.success({
+            message,
+            duration: 10,
+          });
+          dispatch(success(reducerTypes.GENERATE_NOTICE_OF_WORK_APPLICATION_DOCUMENT));
+          if (downloadGeneratedDocument) {
```

Comment:
Is this condition always true? I don't see any changes in this PR that would pass a false param.

Reply:
Yeah that's correct

Reply:
Do we need the if statement then? Planning ahead?